### ISSUE TITLE
Create: GUI helper to bridge GUI and deck

### ISSUE THREAD

**Issue by huiminlim** [AY1920S1-CS2103-F09-2/main#138]
Close issue #137 

TLDR:
![20191025_165022](https://user-images.githubusercontent.com/43434313/67557492-c9175900-f747-11e9-8d9d-749648ffe30c.jpg)

Create a `GUI Helper`, segregate the roles as follows:
`GUI Helper`: extract all the data that `GUI` needs to display, create an `GUI` object for `GUI`
`GUI`: display the GUI object that `GUI helper` produced
Action: `Deck`, `flashcards` will be modified to support `GUI helper`

**Context**
Make code cleaner, separate GUI components from flashcards and deck.

**Problem:**
- Due to polymorphism usage, Deck has no knowledge of what type is flashcard
- We do not want to check instance of flashcard (i.e. == mcq, frontback etc) each time in GUI helper

**Solution**
- Create a `GUI helper`.
- Prepare an object to display in `GUI`
- `GUI helper` call getter methods of flashcard interface.
- For choices in MCQ cards, use a boolean method call to check if Flashcard has choices attached, then call getChoice() if return true

**Comment by nattanyz** [comment]
can't call `getListOfChoices()` even after checking `hasChoices()` because `getListOfChoices()` is not defined in `FlashCard`. maybe you need to add it to the abstract class? :O @huiminlim

**Comment by huiminlim** [comment]
![image](https://user-images.githubusercontent.com/43434313/67636900-41c91180-f910-11e9-8f2f-88928c74d7bb.png)

**Comment by nattanyz** [comment]
seems like it should work, but when should I call `shuffleChoices()` then?
if I add to abstract class, should the other classes that extend from `FlashCard` just return an empty arraylist?
